src/main/java/com/intuit/graphql/orchestrator/authorization/FieldAuthorization.java

```diff
@@ -0,0 +1,16 @@
+package com.intuit.graphql.orchestrator.authorization;
+
+import com.intuit.graphql.orchestrator.common.FieldPosition;
+import graphql.GraphqlErrorException;
+import java.util.concurrent.CompletableFuture;
+
+public interface FieldAuthorization<AuthDataT> {
+
+    boolean isAccessAllowed(FieldAuthorizationRequest<AuthDataT> fieldAuthorizationRequest);
+
+    boolean requiresAccessControl(FieldPosition fieldPosition);
+
+    CompletableFuture<AuthDataT> getFutureAuthData();
+
+    GraphqlErrorException getDeniedGraphQLErrorException();
```

Comment:
Any reason for this method to be implemented by the class overwriting it.

Reply:
The objective is to 
- allow the implementing class create custom GraphQLErrorExceptions is required by the application e.g. adding error codes in `error.extenstions`
- ensure that the error is thrown.  Was thinking if the actual throwing of exception should be performed by the implement class or the BatchLoader/Query Redactor.